src/sentry/logging/handlers.py

```diff
@@ -19,6 +19,18 @@ def emit(self, record):
         }
         if record.exc_info:
             kwargs['exc_info'] = record.exc_info
+
+        if '%' in record.msg and record.args:
+            try:
+                populated = record.msg % record.args
+            except TypeError:
```

Comment:
A `KeyError` could also be raised.

You can just catch everything with `except Exception` and not worry about it.